web/src/main/java/se/inera/intyg/webcert/web/service/monitoring/MonitoringLogServiceImpl.java

```diff
@@ -241,57 +519,167 @@ public void logPrivatePractitionerTermsApproved(String userId, Personnummer pers
+        final var subject = amne != null ? amne.name() : NO_AMNE;
         if (isAnswer) {
-            logEvent(MonitoringEvent.ARENDE_CREATED_ANSWER, amne != null ? amne.name() : "NO AMNE", intygsId, intygsTyp, unitHsaId);
+            try (MdcCloseableMap ignored = MdcCloseableMap.builder()
+                .put(MdcLogConstants.EVENT_ACTION, toEventType(MonitoringEvent.ARENDE_CREATED_ANSWER))
+                .put(MdcLogConstants.EVENT_TYPE, MdcLogConstants.EVENT_TYPE_CREATION)
+                .put(MdcLogConstants.EVENT_MESSAGE_SUBJECT, subject)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_ID, intygsId)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_TYPE, intygsTyp)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_UNIT_ID, unitHsaId)
+                .build()
+            ) {
+                logEvent(MonitoringEvent.ARENDE_CREATED_ANSWER, subject, intygsId, intygsTyp, unitHsaId);
+            }
         } else {
-            logEvent(MonitoringEvent.ARENDE_CREATED_QUESTION, amne != null ? amne.name() : "NO AMNE", intygsId, intygsTyp, unitHsaId);
+            try (MdcCloseableMap ignored = MdcCloseableMap.builder()
+                .put(MdcLogConstants.EVENT_ACTION, toEventType(MonitoringEvent.ARENDE_CREATED_QUESTION))
+                .put(MdcLogConstants.EVENT_TYPE, MdcLogConstants.EVENT_TYPE_CREATION)
+                .put(MdcLogConstants.EVENT_MESSAGE_SUBJECT, subject)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_ID, intygsId)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_TYPE, intygsTyp)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_UNIT_ID, unitHsaId)
+                .build()
+            ) {
+                logEvent(MonitoringEvent.ARENDE_CREATED_QUESTION, subject, intygsId, intygsTyp, unitHsaId);
+            }
         }
     }
 
     @Override
     public void logIntegratedOtherUnit(String intygsId, String intygsTyp, String unitId) {
-        logEvent(MonitoringEvent.LOGIN_OTHER_UNIT, intygsId, intygsTyp, unitId);
+        try (MdcCloseableMap ignored = MdcCloseableMap.builder()
+            .put(MdcLogConstants.EVENT_ACTION, toEventType(MonitoringEvent.LOGIN_OTHER_UNIT))
+            .put(MdcLogConstants.EVENT_TYPE, MdcLogConstants.EVENT_TYPE_USER)
+            .put(MdcLogConstants.EVENT_CERTIFICATE_ID, intygsId)
+            .put(MdcLogConstants.EVENT_CERTIFICATE_TYPE, intygsTyp)
+            .put(MdcLogConstants.ORGANIZATION_ID, unitId)
+            .build()
+        ) {
+            logEvent(MonitoringEvent.LOGIN_OTHER_UNIT, intygsId, intygsTyp, unitId);
+        }
     }
 
     @Override
     public void logIntegratedOtherCaregiver(String intygsId, String intygsTyp, String caregiverId, String unitId) {
-        logEvent(MonitoringEvent.LOGIN_OTHER_CAREGIVER, intygsId, intygsTyp, caregiverId, unitId);
+        try (MdcCloseableMap ignored = MdcCloseableMap.builder()
```

Comment:
Would be great if we logged event.certificate.unit.id and event.certificate.care_provider.id. They need to be added as arguments. They are valuable to when reading this log event.